ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -290,23 +249,49 @@ public boolean updateAccounts(Collection<Account> accounts) {
         accountServiceMetrics.notifyAccountDataChangeErrorCount.inc();
       }
     } else {
-      logger.error("Failed updating accounts={}", accounts);
+      logger.error("Failed updating accounts={}, took {} ms", accounts, timeForUpdate);
       accountServiceMetrics.updateAccountErrorCount.inc();
     }
     return hasSucceeded;
   }
 
   @Override
   public void close() {
-    if (isOpen) {
-      isOpen = false;
+    if (open.compareAndSet(true, false)) {
       if (scheduler != null) {
-        shutDownExecutorService(scheduler, storeConfig.accountUpdaterShutDownTimeoutMs, TimeUnit.MILLISECONDS);
+        shutDownExecutorService(scheduler, config.updaterShutDownTimeoutMs, TimeUnit.MILLISECONDS);
       }
       helixStore.stop();
     }
   }
 
+  /**
+   * To be used to subscribe to a {@link Notifier} topic. Upon receiving a
+   * {@link #FULL_ACCOUNT_METADATA_CHANGE_MESSAGE}, it will check for any account updates.
+   * @param topic The topic.
+   * @param message The message for the topic.
+   */
+  private void onAccountChangeMessage(String topic, String message) {
+    checkOpen();
```

Comment:
looks like this has no changes - just a refactoring. Correct?

Reply:
Yes, I just was trying to move code out of the constructor